source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -14,7 +14,6 @@ ListenerFilterBufferImpl::ListenerFilterBufferImpl(IoHandle& io_handle,
       on_data_cb_(on_data_cb), buffer_(std::make_unique<uint8_t[]>(buffer_size)),
       base_(buffer_.get()), buffer_size_(buffer_size) {
   // If the buffer_size not greater than 0, it means that doesn't expect any data.
-  ASSERT(buffer_size > 0);
```

Comment:
what will return for the `recv` syscall if you pass a 0 size?

Reply:
In summary, we have 2 options

1. set buffer size to 1 if maxReadBytes is 0
2. keep buffer size to 0, and only enable Close event for it

Option 2 looks like an elegant solution, but it is limited by the fact that on macOS the connection doesn't get early close notification and instead gets the close after reading the FIN (verified), and the Windows backend in libevent does not support the EV_CLOSED flag. So Enabling Read event is a must for cross-platform support. And on macOS if Read is enabled with 0 buffer size, the connection will be closed when there is any data sent. So finally I decided to go with set buffer size to 1.

Reply:
> In summary, we have 2 options
> 
> 1. set buffer size to 1 if maxReadBytes is 0
> 2. keep buffer size to 0, and only enable Close event for it
> 
> Option 2 looks like an elegant solution, but it is limited by the fact that on macOS the connection doesn't get early close notification and instead gets the close after reading the FIN (verified), and the Windows backend in libevent does not support the EV_CLOSED flag. So Enabling Read event is a must for cross-platform support. And on macOS if Read is enabled with 0 buffer size, the connection will be closed when there is any data sent. So finally I decided to go with set buffer size to 1.

Thanks for the verification!